### PR TITLE
CNI, HA and Cleanup

### PR DESCRIPTION
**Pull request by appkins**
* Adds support for HA
* Adds Support for the CNI plugin (removes one of the side cars from linkerd init)
* Moves cert manager resources to kubernetes provider

### CODE REVIEW

File: modules/issuer/main.tf [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  }
+}
+
+
+# create namespaces for linkerd and any extensions (linkerd-viz or linkerd-jaeger)
+resource "kubernetes_namespace" "namespace" {
+  for_each = toset(local.namespaces)
+  metadata {
+    name        = each.key
+    annotations = (each.key != var.namespace) ? { "linkerd.io/inject" = "enabled" } : {}
```
Comment (ggolub) [inline modules/issuer/main.tf:31]: I know this is legacy, but not sure why we are setting the "inject" annotation on the linkerd namespaces themselves.

**Review by ggolub — COMMENTED** [review]
Serious rework but much improved.  I am humbled.

File: modules/issuer/main.tf (hunk 1 above)
Comment (ggolub) [reply modules/issuer/main.tf:31]: @appkins - Some of the checks are failing.  Thoughts?

Comment (ggolub) [reply modules/issuer/main.tf:31]: @appkins Did you test stopping and restarting the cluster with HA installed?

Comment (ggolub) [reply modules/issuer/main.tf:31]: When running "linkerd check" I see an issue with the certs:
` 1m×× tap API server has valid cert
    cert is not issued by the trust anchor: x509: certificate is valid for tap.linkerd.svc, not tap.linkerd-viz.svc
    see https://linkerd.io/checks/#l5d-tap-cert-valid for hints`

**Review by ggolub — CHANGES_REQUESTED** [review]
Comments added.  Of particular concern is HA mode which I can't get working.  Happy to collaborate on this.

File: modules/issuer/main.tf (hunk 1 above)
Comment (appkins) [reply modules/issuer/main.tf:31]: I made a bunch of changes last week @ggolub 
I was seeing the same issues, but now am able to deploy HA linkerd that works with vault injector. The latest changes move to the edge release of linkerd that provides two helm charts for the deployment. Linkerd now has a crd deployment and a core component. This allows us to deploy linkerd crds with no dependencies, create custom resources, then deploy the meat of linkerd. The deployments seem to be much more stable after the last batch of changes.

Comment (appkins) [reply modules/issuer/main.tf:31]: Another thing worth mentioning, linkerd inject annotations are just an effect of the namespace selectors provided in the helm chart. I'm now overriding the namespace selector for proxyinjector which was causing issues before on linkerd namespaces.

```terraform
proxyInjector = {
      namespaceSelector = {
        matchExpressions = [
          { key = "linkerd.io/is-control-plane", operator = "DoesNotExist" },
          { key = "linkerd", operator = "In", values = ["enabled"] }
        ]
      }
    }
```

This means that a resource (namespace) with the label `linkerd: enabled` will be processed by proxy injector. We have a lot of flexibility and might want to review how we set up the selectors before merging this PR.

Comment (appkins) [reply modules/issuer/main.tf:31]: > I know this is legacy, but not sure why we are setting the "inject" annotation on the linkerd namespaces themselves. I think this was a day-one error that should be removed. We don't need lines 31 and 32

Good idea, we could actually add the annotation `linkerd.io/is-control-plane` and explicitly disable proxy injector for linkerd namespaces, allowing linkerd to be enabled by default everywhere else in the cluster. How do you feel about that solution for a default path?

**Review by riesbl01 — APPROVED** [review]
Havent used this module yet so knowledge may be limited, but syntax wise everything is looking good.

**Pull request merged**